### PR TITLE
[GLM5.3 Perf] Size the GLM-5 sparse indexer decode workspace, 3072 MiB GPU memory saved

### PR DESCRIPTION
**Pull request by yewentao256**
## Purpose

1. Rename `max_model_len` to `max_pool_len` for GLM indexer

2. Size the GLM-5 sparse indexer decode workspace using the pooled sequence length:

`max_pool_len = max_model_len // index_kpool`

This reduces the decode logits workspace without changing model behavior

By default model len `1,048,576` we get 4x memory reduction, namely 3072 MiB memory saved

## Test

Acc covered in current unit test

A sweep through different model len

```py
"""
import statistics

import torch

from vllm.platforms import current_platform
from vllm.utils.deep_gemm import (
    fp8_fp4_paged_mqa_logits,
    get_num_sms,
    get_paged_mqa_logits_metadata,
)
from vllm.utils.import_utils import has_deep_gemm

TOKEN_LENGTHS = (16_384, 65_536, 262_144, 1_048_576)
KPOOL = 4
BATCH = 4
HEADS = 32
HEAD_DIM = 128
BLOCK_SIZE = 64
WARMUP = 5
REPEATS = 50

# Realistic B300 server configuration. These values match vLLM's defaults on
# GPUs with at least 160 GiB of memory.
MAX_NUM_SEQS = 1024
MAX_NUM_BATCHED_TOKENS = 16_384
NUM_SPECULATIVE_TOKENS = 0
PREFILL_WORKSPACE_MIB = 512


def benchmark(run, width):
    for _ in range(WARMUP):
        run(width)
        torch.accelerator.synchronize()

    samples = []
    start = torch.Event(enable_timing=True)
    end = torch.Event(enable_timing=True)
    for _ in range(REPEATS):
        start.record()
        run(width)
        end.record()
        end.synchronize()
        samples.append(start.elapsed_time(end))
    return statistics.median(samples)


def profiled_workspace_mib(width):
    decode_rows = min(
        MAX_NUM_SEQS * (NUM_SPECULATIVE_TOKENS + 1), MAX_NUM_BATCHED_TOKENS
    )
    decode_mib = decode_rows * width * torch.float32.itemsize / 2**20
    return max(decode_mib, PREFILL_WORKSPACE_MIB)


@torch.inference_mode()
def run_case(token_length):
    pool_length = token_length // KPOOL
    num_blocks = pool_length // BLOCK_SIZE
    q = torch.zeros(BATCH, 1, HEADS, HEAD_DIM, dtype=torch.float8_e4m3fn, device="cuda")
    weights = torch.ones(BATCH, HEADS, dtype=torch.float32, device="cuda")
    kv_cache = torch.zeros(
        num_blocks,
        BLOCK_SIZE,
        1,
        HEAD_DIM + 4,
        dtype=torch.uint8,
        device="cuda",
    )
    context_lens = torch.full((BATCH, 1), pool_length, dtype=torch.int32, device="cuda")
    block_tables = (
        torch.arange(num_blocks, dtype=torch.int32, device="cuda")
        .expand(BATCH, -1)
        .contiguous()
    )
    schedule = get_paged_mqa_logits_metadata(context_lens, BLOCK_SIZE, get_num_sms())

    def run(width):
        return fp8_fp4_paged_mqa_logits(
            (q, None),
            kv_cache,
            weights,
            context_lens,
            block_tables,
            schedule,
            width,
            clean_logits=False,
        )

    before = run(token_length)
    after = run(pool_length)
    torch.accelerator.synchronize()
    torch.testing.assert_close(before[:, :pool_length], after, rtol=0, atol=0)

    before_ms = benchmark(run, token_length)
    after_ms = benchmark(run, pool_length)
    return before.nbytes / 2**20, after.nbytes / 2**20, before_ms, after_ms


def main():
    if not torch.accelerator.is_available() or not has_deep_gemm():
        raise RuntimeError("This benchmark requires CUDA and DeepGEMM")

    print(f"GPU: {current_platform.get_device_name()}")
    print(f"batch={BATCH}, index_kpool={KPOOL}, repeats={REPEATS}")
    print(
        "server: "
        f"max_num_seqs={MAX_NUM_SEQS}, "
        f"max_num_batched_tokens={MAX_NUM_BATCHED_TOKENS}, "
        f"num_speculative_tokens={NUM_SPECULATIVE_TOKENS}, "
        f"prefill_workspace={PREFILL_WORKSPACE_MIB} MiB"
    )
    print(
        "tokens     output MiB          latency ms       speedup  "
        "profiled workspace MiB"
    )
    print(
        "            before   after     before   after             "
        "before   after   saved"
    )
    for token_length in TOKEN_LENGTHS:
        before_mib, after_mib, before_ms, after_ms = run_case(token_length)
        workspace_before = profiled_workspace_mib(token_length)
        workspace_after = profiled_workspace_mib(token_length // KPOOL)
        print(
            f"{token_length:>9,}  {before_mib:>8.2f} {after_mib:>7.2f}  "
            f"{before_ms:>8.3f} {after_ms:>7.3f}  "
            f"{before_ms / after_ms:>6.3f}x  "
            f"{workspace_before:>8.0f} {workspace_after:>7.0f} "
            f"{workspace_before - workspace_after:>7.0f}"
        )
        torch.accelerator.empty_cache()


if __name__ == "__main__":
    main()

```

```bash
tokens     output MiB          latency ms       speedup  profiled workspace MiB
            before   after     before   after             before   after   saved
   16,384      0.25    0.06     0.016   0.016   1.019x       512     512       0
   65,536      1.00    0.25     0.016   0.016   1.040x       512     512       0
  262,144      4.00    1.00     0.018   0.018   1.009x      1024     512     512
1,048,576     16.00    4.00     0.024   0.024   1.003x      4096    1024    3072
```

### CODE REVIEW

**Review by sfeng33 — APPROVED** [review]
Thanks, LGTM!
